src/main/kotlin/com/hibob/academy/employeeFeedback/dao/FeedbackDao.kt

```diff
@@ -38,4 +38,34 @@ class FeedbackDao(private val sql: DSLContext) {
             .execute() > 0
     }
 
+
+
+    fun getFeedbackByParams(companyId: Long, params: FeedbackQueryParams): List<FeedbackData> {
+        val baseQuery = sql.select()
+            .from(feedbackTable)
+            .leftJoin(employeeTable).on(feedbackTable.employeeId.eq(employeeTable.id))
+            .where(feedbackTable.companyId.eq(companyId))
+
+        val query = baseQuery
+            .let { q ->
+                params.department?.let {
+                    q.and(employeeTable.department.eq(it.name))
+                } ?: q
+            }
+            .let { q ->
+                if (params.nullEmployeeId) {
+                    q.and(feedbackTable.employeeId.isNull)
+                } else {
+                    q
+                }
+            }
+            .let { q ->
+                params.date?.let {
+                    q.and(feedbackTable.date.eq(it))
+                } ?: q
+            }
```

Comment:
i like! let's just extract each let to a method, like handleDepartmentCondition, handleEmployeeIdCondition etc